src/main/kotlin/org/rust/cargo/project/model/impl/CargoEventService.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.model.impl
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import org.rust.cargo.project.model.CargoProjectsService.CargoProjectsListener
+import org.rust.cargo.project.model.CargoProjectsService.Companion.CARGO_PROJECTS_TOPIC
+import org.rust.cargo.runconfig.command.workingDirectory
+import org.rust.stdext.mapToSet
+import java.nio.file.Path
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+@Service
+class CargoEventService(project: Project) {
```

Comment:
Let's add a short comment about these hacks. 
By the way, maybe just move `metadataCallTimestamps` to CargoProjectsServiceImpl?

Reply:
> By the way, maybe just move metadataCallTimestamps to CargoProjectsServiceImpl?

It will lead to complication of `CargoProjectsServiceImpl` itself and using `CargoProjectsServiceImpl` in some additional places that I think we should try to avoid. Of course, we can add some method/property into `CargoProjectsService` but I consider that these changes are too specific and I'd prefer to keep them separately from `CargoProjectsService`